samples/README.md

```diff
@@ -18,3 +22,135 @@ for that code, and they always have the option not to.
 The hope is that samples like these may be useful to those who wish to discuss
 the spec or implement tools. To that end, we welcome comments on both the format
 of these files and their content.
+
+## Directory structure
+
+See
+[jspecify: test-data format: Directory structure](https://docs.google.com/document/d/1JVH2p61kReO8bW4AKnbkpybPYlUulVmyNrR1WRIEE_k/edit#bookmark=id.2t1r58i5a03s).
+TODO: Inline that here if Tagir can sign the CLA and contribute it.
+
+Additionally:
+
+Fully qualified class names must be unique across all directories.
+
+> This permits all files to be compiled in a single tool invocation.
+
+TODO: Consider requiring that all individual-file samples be in the top-level
+directory.
+
+TODO: Consider requiring a file's path to match its package and class:
+
+-   individual-file samples: `Foo.java` for `Foo`
```

Comment:
Do we allow multiple package-local classes in one file? It's usually very convenient for tests

Reply:
In most of the cases, it's possible to create nested classes. I think it's better (have a single top-level class and place all other classes inside), as makes it easier to avoid the name clash. If some test absolutely requires several top-level classes, it's also should be ok, though probably some naming convention (e.g. common prefix) would be nice.